server/metadata/tenant.go

```diff
@@ -301,6 +350,41 @@ func (m *TenantManager) GetTenant(ctx context.Context, namespaceId string) (*Ten
 	return tenant, nil
 }
 
+// UpdateNamespaceMetadata updates a namespace metadata for a tenant
+func (m *TenantManager) UpdateNamespaceMetadata(ctx context.Context, meta NamespaceMetadata) error {
+	var (
+		err error
+	)
+
+	m.Lock()
+	defer m.Unlock()
+
+	tx, err := m.txMgr.StartTx(ctx)
+	if err != nil {
+		return err
+	}
+
+	if _, found := m.tenants[meta.StrId]; !found {
+		return errors.NotFound("no tenant found with id %s", meta.StrId)
+	}
+
+	defer func() {
+		if err == nil {
+			tenant := m.tenants[meta.StrId]
+			if err = tx.Commit(ctx); err == nil && tenant != nil {
+				tenant.Lock()
+				tenant.namespace = NewTenantNamespace(meta.StrId, meta)
```

Comment:
Just update the accounts or the metadata?

Reply:
`tenant.namespace` is an interface, we cannot replace the metadata:

https://github.com/tigrisdata/tigris/blob/bb8cad820de6c98b3456ebc3dd1bf35dbe6cee59/server/metadata/namespace.go#L27-L34

Reply:
We can have`SetMetadata(md NamespaceMetadata)` on it.